File: src/main/java/org/chillout1778/MotorSpin.java
package org.chillout1778;

import com.revrobotics.CANSparkMax;
import com.revrobotics.CANSparkMaxLowLevel;
import edu.wpi.first.wpilibj2.command.SubsystemBase;
import edu.wpi.first.math.controller.PIDController;
import edu.wpi.first.math.util.Units;

public class MotorSpin extends SubsystemBase {
    private CANSparkMax motor = new CANSparkMax(11,
            CANSparkMaxLowLevel.MotorType.kBrushless);

    private static final double kP = 1.0;
    private static final double kI = 0.0;
    private static final double kD = 0.0;

    private PIDController controller = new PIDController(kP, kI, kD);
    private double desired = 0.0;  // rad/s

    public void spin() {
        desired = 0.1;
    }

    public void stop() {
        desired = 0.0;
    }

    private double angularVelocity() {
        return Units.rotationsPerMinuteToRadiansPerSecond(
                motor.getEncoder().getVelocity()
        );
    }

    @Override
    public void periodic() {
        double voltage = controller.calculate(angularVelocity(), desired);
    }
}

File: src/main/java/org/chillout1778/subsystems/MotorSpin.kt
package org.chillout1778.subsystems

import com.revrobotics.CANSparkMax
import com.revrobotics.CANSparkMaxLowLevel
import edu.wpi.first.math.controller.PIDController
import edu.wpi.first.math.util.Units
import edu.wpi.first.wpilibj2.command.SubsystemBase

object MotorSpin : SubsystemBase() {
    private const val kP = 1.0
    private const val kI = 0.0
    private const val kD = 0.0
    private val motor = CANSparkMax(
        11,
        CANSparkMaxLowLevel.MotorType.kBrushless
    )
    private val controller = PIDController(kP, kI, kD)
    private var desired = 0.0 // rad/s
    fun spin() {
        desired = 0.1
    }

    fun stop() {
        desired = 0.0
    }

    private fun angularVelocity(): Double {
        return Units.rotationsPerMinuteToRadiansPerSecond(
            motor.getEncoder().velocity
        )
    }

    override fun periodic() {
        val voltage = controller.calculate(angularVelocity(), desired)
        motor.setVoltage(voltage)
    }
}
